internal/config/instance.go

```diff
@@ -216,39 +242,74 @@ func (i *Instance) InstallSource() string {
 
 // ReadInConfig reads in config from the config file
 func (i *Instance) ReadInConfig() error {
-	i.rwLock.RLock()
-	defer i.rwLock.RUnlock()
-	// Prepare viper, which is a library that automates configuration
-	// management between files, env vars and the CLI
-	i.viper.SetConfigName(i.Name())
-	i.viper.SetConfigType(i.Type())
-	i.viper.AddConfigPath(i.configDir.Path)
-	i.viper.AddConfigPath(".")
-
-	if err := i.viper.ReadInConfig(); err != nil {
-		return errs.Wrap(err, "Cannot read config.")
+	configData, err := i.configFileData()
```

Comment:
Why do we have a separate `configFileData()` function? Any reason why we'd be reading the config and not populating `i.data`? I can see in the `Set()` function you are updating it right away, but it doesn't cost much to populate a variable twice in a row.

Reply:
You're right this function isn't necessary.